bookwyrm/activitystreams.py

```diff
@@ -111,8 +111,18 @@ def get_audience(self, status):  # pylint: disable=no-self-use
             Q(id__in=status.user.blocks.all()) | Q(blocks=status.user)  # not blocked
         )
 
+        # don't show replies to statuses the user can't see
+        if status.reply_parent and status.reply_parent.privacy == "followers":
+            audience = audience.filter(
+                Q(id=status.user.id)  # if the user is the post's author
+                | Q(id=status.reply_parent.user.id)  # if the user is the OG author
+                | (
+                    Q(following=status.user) & Q(following=status.reply_parent.user)
+                )  # if the user is following both authors
+            ).distinct()
+
         # only visible to the poster and mentioned users
-        if status.privacy == "direct":
+        elif status.privacy == "direct":
```

Comment:
If the parent status is a followers-only status, and then a reply is sent as a DM, will this mean that the DM is visible to followers, rather than only visible to tagged users?

Reply:
Hmm good question.

I *think* I tested this exact scenario and it wasn't a problem, but it might have been the other way around (a followers-only response to a DM) so I'll double check.

`HomeStream` excludes DMs so I think that prevents it, but for a belt and braces approach we could probably do:

```python
if status.reply_parent.privacy == "followers" and status.privacy not "direct" :
```

Reply:
I've checked this and it all seems to be ok. However to make it clearer, I changed the order so that DMs are at top, `elif` the parent post is followers-only... etc.

I did discover a privacy problem whilst checking this, but it wasn't created in this PR, and I'm a bit unsure how to fix it. I will add another issue for that.